h11/_receivebuffer.py

```diff
@@ -83,30 +91,56 @@ def maybe_extract_until_next(self, needle):
         # Returns extracted bytes on success (advancing offset), or None on
         # failure
         if self._looked_for == needle:
-            search_start = max(self._start, self._looked_at - len(needle) + 1)
+            looked_at = max(self._start, self._looked_at - len(needle) + 1)
```

Comment:
As `len(len(b"\n\r?\n") == 4` (rather than 3) isn't this incorrect now? 

Although I think it should work as long as the regex representation is longer than the actual matchable length (which may not be the case)?

Reply:
Also I think we need to note that `needle` is now compiled (wasn't before) in say the docstring.

Alternatively I think this method doesn't need to take a `needle` parameter or be as generic. Maybe it is worth rewriting as `maybe_extract_until_next_line_delimiter` (or similar).

Reply:
> Although I think it should work as long as the regex representation is longer than the actual matchable length (which may not be the case)?

Well, yes, thinking about your comment, I found that `.*` might be much wider that 2 symbols, so I was definitely wrong in my assumption coding that

Reply:
So what I did regarding this remark:
1. I changed signature of the `maybe_extract_until_next`. Now it's
```python
def maybe_extract_until_next(self, needle_regex, max_needle_length):
```
2. it accepts pre-compiled regex and max needle length, for example:
```python
data = self.maybe_extract_until_next(blank_line_delimiter_regex, 3)
if data is None:
    return None
```
or
```python
if self._bytes_in_chunk == 0:
    # We need to refill our chunk count
    chunk_header = buf.maybe_extract_until_next(line_delimiter_regex, 2)
    if chunk_header is None:
        return None
```